core/include/scipp/core/element/arithmetic.h

```diff
@@ -14,13 +14,26 @@ namespace scipp::core::element {
 
 constexpr auto add_inplace_types =
     arg_list<double, float, int64_t, int32_t, Eigen::Vector3d,
+             std::tuple<scipp::core::time_point, int64_t>,
+             std::tuple<scipp::core::time_point, int32_t>,
              std::tuple<double, float>, std::tuple<int64_t, int32_t>,
              std::tuple<int64_t, bool>>;
 
-constexpr auto plus_equals =
-    overloaded{add_inplace_types, [](auto &&a, const auto &b) { a += b; }};
-constexpr auto minus_equals =
-    overloaded{add_inplace_types, [](auto &&a, const auto &b) { a -= b; }};
+constexpr auto plus_equals = overloaded{
+    add_inplace_types,
+    [](auto &&a, const auto &b) { a += b; },
+    [](scipp::core::time_point &a, const auto &b) {
+      a += std::chrono::duration<int64_t>(b);
```

Comment:
Another problem here that I had previously overlooked: `std::chrono::system_clock` is probably in seconds not nanoseconds? It appears to be unspecified though. However, the default for `duration` *is* specified, and it is seconds (if I understood cppreference correctly), so this is not what we need. Can be solved alongside the changes we will likely have to to relating to `time_point`.